### DISCUSSION TITLE
Type Tool:  Make text input non-destructive and editable

### DISCUSSION THREAD

**Discussion by ghost** [opentoonz/opentoonz#6647] [General]
**Is your feature request related to a problem? Please describe**

Make the text editable with the type tool. This would be a great help when animating texts with vector styles when you make mistakes or in cases of duplicating effects with other texts. Unfortunately, the text iwa effect does not allow to apply styles.

**Comment by RodneyBaker** [comment]
The desire for editable text inside of Opentoonz remains.
As such this request should be captured in the development roadmap although the need for editable text is somewhat obvious.

In the meantime, there are a number of ways to workaround and with the Type Tool's lack of text editablility (none being optimal):

- Use and store editable text in Memo or Notelevel and transfer to Type Tool
- Use Text Iwa FX with or without Notelevels instead of Type Tool
- Use Clapperboard's text input for frames requiring editable text (Set up a scene with intention of just using Clapperboard for this purpose)

Bottom Line:  It can help to understand that the Type Tool is not a Text Tool.  The type is like old Guttenburg letter presses which once set down in the mold are not non destructively editable.  Opentoonz does not have a Text Tool but does have some textual tools in a variety of locations.

**Comment by RodneyBaker** [comment]
Regarding the Type Tool's ability to store and retrieve editable text, one approach might be to have the Type Tool get it's input from a text file.  That text file then would be updateable each time the Type Tool is used.  The type created in an image itself would not be editable but the text would be preserved so that it could be reused and updated.  A dropdown menu might even allow for selection of previously typed text.

All this to say the Type Tool should be reviewed in light of current and intended purpose.
It can be improved considerably within the paradigm of 'Type Tool'.

A Text Tool will likely need to be developed to gain full editable text support.

**Comment by RodneyBaker** [comment]
Converting to discussion but this is a logical extension of where editable text in Opentoonz needs to go.

Short of adding an actual Text Tool I'd say we might be able to adjust the Type Tool so that when activated a dialogue opens which references a text file where it stores the textual entry.  The Type Tool would then use that text to generate the vector glyphs.

Two general questions:
What should success with the Type Tool look like?
What would success look like with a Text Tool?
